### PR TITLE
Replace perf_client usage with unit tests

### PR DESCRIPTION
**Pull request by rmccorm4**
Use infer_util unit tests in place of perf_analyzer wherever possible to reduce PA dependency

### CODE REVIEW

File: qa/L0_storage_azure/test.sh [hunk 1]
```diff
@@ -179,20 +185,16 @@ if [ "$SERVER_PID" == "0" ]; then
 fi
 
 set +e
-for BACKEND in $BACKENDS; do
-    code=`curl -s -w %{http_code} -X POST localhost:8000/v2/repository/models/${BACKEND}_float32_float32_float32/load`
+for model in `ls models/`; do
```
Comment (CoderHam) [inline qa/L0_storage_azure/test.sh:188]: Why is this change needed?

File: qa/L0_storage_azure/test.sh [hunk 2]
```diff
@@ -202,46 +204,59 @@ wait $SERVER_PID
...
-done
-sleep 10
-
-run_server
-if [ "$SERVER_PID" == "0" ]; then
-    echo -e "\n***\n*** Failed to start $SERVER\n***"
-    cat $SERVER_LOG
-    exit 1
-fi
+# FIXME: DLIS-3893 - Currently using autocomplete will ignore the label_filename
```
Comment (CoderHam) [inline qa/L0_storage_azure/test.sh:207]: We need these tests. If we cannot work around this limitation yet, I'd recommend holding off on adding these changes. You do not need label file name for non-classification output for example.  

Since these tests are our only way to validate cloud storage support, disabling them temporarily would cause testing gaps.

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: qa/L0_storage_azure/infer_test.py [hunk 3]
```diff
@@ -78,38 +79,52 @@ def _infer_exact_helper(tester,
                                use_streaming=use_streaming,
                                correlation_id=correlation_id)
 
+        # Allow caller to setup subset of backends to test
+        test_backends = os.environ.get("INFER_TEST_BACKENDS", "").split()
```
Comment (CoderHam) [inline qa/L0_storage_azure/infer_test.py:83]: Set this above under as TEST_BACKENDS.

File: qa/L0_storage_azure/test.sh [hunk 4]
```diff
@@ -83,18 +83,38 @@ rm -f $SERVER_LOG_BASE* $CLIENT_LOG_BASE*
 RET=0
 
 BACKENDS="graphdef savedmodel onnx libtorch plan"
```
Comment (CoderHam) [inline qa/L0_storage_azure/test.sh:85]: BACKENDS=${BACKENDS:="graphdef savedmodel onnx libtorch plan"}

File: qa/L0_storage_azure/infer_test.py [hunk 5]
```diff
@@ -37,6 +37,7 @@
 
 np_dtype_string = np.dtype(object)
 
+DEFAULT_TEST_BACKENDS = ["graphdef", "savedmodel", "plan", "onnx", "libtorch"]
```
Comment (CoderHam) [inline qa/L0_storage_azure/infer_test.py:40]: BACKENDS = os.environ.get(
    'BACKENDS',
    "graphdef savedmodel onnx libtorch plan")

**Review by CoderHam — CHANGES_REQUESTED** [review]

**Comment by rmccorm4** [comment]
I think it makes sense to wait for autocomplete changes to be made first rather than adding throwaway work or temporarily disabling part of the test. Converting to draft for now.

File: qa/L0_storage_azure/test.sh (hunk 1 above)
Comment (rmccorm4) [reply qa/L0_storage_azure/test.sh:187]: I'm now loading everything in the model dir, not just the float32 models, in order to run the unit tests which also test the `*_object` versions of the models.

**Comment by rmccorm4** [comment]
Autocomplete changes were merged and these pass in CI now.

**Comment by rmccorm4** [comment]
- Refactored `L0_s3_local` to use `infer_test.py`
- Moved duplicate `L0_storage_*/infer_test.py` into `qa/common/infer_test.py` where applicable (used in gitlab tests too)

**Review by tanmayv25 — APPROVED** [review]

**Comment by rmccorm4** [comment]
Passing CI :heavy_check_mark:

File: qa/L0_s3_local/test.sh [hunk 6]
```diff
@@ -41,17 +41,63 @@ fi
...
+# Used to control which backends are run in infer_test.py
+BACKENDS=${BACKENDS:="graphdef savedmodel onnx libtorch plan"}
 
-rm -rf models && mkdir models
-for BACKEND in $BACKENDS; do
-    cp -r $DATADIR/${BACKEND}_float32_float32_float32 models/.
-    # Remove version policy from config.pbtxt
-    sed -i '/^version_policy/d' models/${BACKEND}_float32_float32_float32/config.pbtxt
-done
+function run_unit_tests {
```
Comment (tanmayv25) [inline qa/L0_s3_local/test.sh:52]: I thing the keyword function has been deprecated in favor of fname(). However, in rest of the project we write the bash script like: `function fname() {}`. 
See here: https://github.com/triton-inference-server/server/blob/main/qa/common/util.sh#L34

Let's follow the same.

**Review by tanmayv25 — APPROVED** [review]

**Comment by rmccorm4** [comment]
Closing and opening new PR to WAR "request changes" block.
